src/snowflake/snowpark/mock/plan.py

```diff
@@ -167,6 +185,90 @@ def __init__(
         self.options = options
 
 
+def handle_order_by_clause(
+    order_by: List[Expression],
+    result_df: TableEmulator,
+    analyzer: "MockAnalyzer",
+    expr_to_alias,
+):
+    sort_columns_array = []
+    sort_orders_array = []
+    null_first_last_array = []
+    if order_by:
+        for exp in order_by:
+            sort_columns_array.append(analyzer.analyze(exp.child, expr_to_alias))
+            sort_orders_array.append(isinstance(exp.direction, Ascending))
+            null_first_last_array.append(
+                isinstance(exp.null_ordering, NullsFirst)
+                or exp.null_ordering == NullsFirst
+            )
+        kk = reversed(
+            list(zip(sort_columns_array, sort_orders_array, null_first_last_array))
+        )
+        for column, ascending, null_first in kk:
+            comparator = partial(custom_comparator, ascending, null_first)
+            result_df = result_df.sort_values(by=column, key=comparator)
+    return result_df
+
+
+def handle_range_frame_indexing(
+    window_spec,
```

Comment:
what does `window_spec` contain?
I see only `window_spec.order_spec` is used in this func, does that mean to get windows we only need `order_spec`?

Reply:
`window_spec` is a `WindowSpec` instance. In this particular function only `order_spec` is being used because`range_between` window frames use the expressions in the order by clause to select the group of rows in the window.